float64[] p_a_s_g
int32 goal_id